### DISCUSSION TITLE
[Android] Inquiry on model rotation/orientation when using Filament with ARCore

### DISCUSSION THREAD

**Discussion by Fe-56** [google/filament#6349] [Q&A] [answered]
Hello all,

I am currently working on an Android AR app that uses both Filament and ARCore. However, I have problems trying to get vertically-placed models (models that are to be placed onto vertical surfaces) to work properly. My AR app is based on this [app](https://github.com/zirman/arcore-filament-example-app).

From my observations, it seems like Filament sets the world position (correct me if I am not using the right terms) at the moment when the Filament engine is first created. Then any model (.glb, etc.) that is being rendered by Filament will be oriented in the direction of that initial world position.

However, this poses an issue in my AR app. This meant that models that are supposed to be placed onto vertical planes may not always be oriented such that they look like they are actually attached to the detected vertical surface. Instead, they are always oriented/pointed towards the opposite direction of where my Android device's camera is pointed towards (i.e. where my device screen is facing) at the moment the Filament engine is created. 

Therefore, this means that, unless I position and orientate my device camera to be facing directly at a vertical surface at a 90 degree angle, the rendered model (by Filament) will always be rotated/oriented away from the detected vertical surface.

I would like to know if there is any way for Filament to render the model such that it will be oriented correctly onto ARCore's detected plane/surface, without doing any vector mathematics or using the Filament engine's `transformManager` to manually rotate the model about a certain axis?

Please let me know if you are unable to understand what I mean, I could provide some screenshots or diagrams for visualisation purposes.

Thank you!

**Comment by romainguy** [comment, marked as answer]
Filament has no orientation with respect to your phone's camera or the real world. That's why AR APIs provide transforms that need to be applied to the virtual camera and/or to objects anchored in the world so you can apply those transforms and track the phone's orientation. You unfortunately can't get around that.

If you want to use an API that makes this easier you can look at https://github.com/SceneView/sceneview-android (this library combines arcore and filament).

**Reply by Fe-56** [reply]
I see, guess I'll have to see how I can make better use of the transforms from the AR APIs then.

Thank you for the quick reply!
